controllers/affiliation_controller.js

```diff
@@ -0,0 +1,76 @@
+const Affiliation = require("../models/affiliation_model.js");
+const Ambassador = require("../models/ambassador_model.js");
+const { SchemaTypes, Schema } = require("mongoose");
+
+module.exports.createAffiliation = async (req, res, next) => {
+  try {
+    const { AmbassadorId } = req.body;
+
+    // Génération du code unique pour le lien d'affiliation
+    const affiliateCode = Math.random().toString(36).substring(2);
```

Comment:
@dev-scott 

An issue can arise here.  It can arrive that, the Math's random function generates 2 similar random codes, therefore crashing as when creating an ```Affiliation``` with a generated code, another ```Affiliation``` could already have that code.
To prevent that situation, I suggest that after you generate the random code, first check if an ```Affiliation``` with the same code exists, if not then create ```Affiliation``` else regenerate until there is no already existing ```Affiliation``` having the same code.